3.0: Improve documentation

rdimaio opened the pull request:
### Description of changes
Small improvements to the code documentation.

I noticed that you are using `:return:` to document what a function returns, but many functions were missing the second ":" - feel free to let me know whether that's intended, in that case I can remove those changes from the pull request.

### Tests
No test functionalities were modified

### References
No related PRs

### Checklist
- [x] Make sure you are pointing to **the right branch** and add a label in the PR title (i.e. **2.x** vs **3.x**)
- [x] Check all commits' messages are clear, describing what and why vs how.
- [x] Make sure **to have added unit tests or integration tests** to cover the new/modified code.
- [x] Check if documentation is impacted by this change.

Please review the [guidelines for contributing](../CONTRIBUTING.md) and [Pull Request Instructions](https://github.com/aws/aws-parallelcluster/wiki/Git-Pull-Request-Instructions).

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

rdimaio commented on cli/src/pcluster/aws/ec2.py:
```diff
@@ -312,8 +312,8 @@ def get_supported_az_for_instance_type(self, instance_type: str):
         """
         Return a tuple of availability zones that have the instance_type.
 
-        This function build above _get_supported_az_for_instance_types,
```
I could not find a function called `_get_supported_az_for_instance_types`, but I saw that `get_supported_az_for_instance_types` (without the underscore at the beginning) is defined just below this function. The version without the initial underscore is the function that's actually called within `get_supported_az_for_instance_type`, so I assume this is the function that was actually meant in the comment

lukeseawalker replied:
correct

rdimaio commented on cli/src/pcluster/aws/ec2.py:
```diff
@@ -312,8 +312,8 @@ def get_supported_az_for_instance_type(self, instance_type: str):
         Return a tuple of availability zones that have the instance_type.
 
-        This function build above _get_supported_az_for_instance_types,
-        but simplify the input to 1 instance type and result to a list
```
As far as I understood, this function returns a tuple and not a list, but feel free to correct this if it's wrong

lukeseawalker replied:
correct

lukeseawalker commented:
@rdimaio can you please fix the check
../tests/integration-tests/s3_common_utils.py:40:121: E501 line too long (122 > 120 characters)
Thanks

rdimaio committed:
Improve documentation

rdimaio commented on cli/src/pcluster/utils.py:
```diff
@@ -155,7 +155,7 @@ def to_iso_timestr(time_in: datetime.datetime) -> str:
 
     :param time_in: datetime to be converted
-    :return time in ISO 8601 UTC format with ms (e.g. 2021-07-15T01:22:02.655Z)
+    :return: time in ISO 8601 UTC format with ms (e.g. 2021-07-15T01:22:02.655Z)
```
Fixed another :return: here in the latest commit that fixes the linter error - commenting for reference

rdimaio commented:
Thanks, fixed the linter error and rebased the new commit into the previous one to keep the commit history clean. I fixed another `:return:` comment in `cli/src/pcluster/utils.py`, added a GH comment to it for reference.
